intro_to_ada.rst

```diff
@@ -3479,32 +3479,802 @@ Predefined exceptions
+package. In the example below, we declare a task ``T`` in the package
+``Simple_Sync_Pkg``.
+
+.. code-block:: ada
+
+    --% src_file: Simple_Sync_Pkg.ads
+    --% cflags: -gnaty
+
+    package Simple_Sync_Pkg is
+       task T;
+    end Simple_Sync_Pkg;
+
+This is the corresponding package implementation:
+
+.. code-block:: ada
+
+    --% src_file: Simple_Sync_Pkg.adb
+    --% cflags: -gnaty
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    package body Simple_Sync_Pkg is
+       task body T is
+       begin
+          for I in 1 .. 10 loop
+             Put_Line ("hello");
+          end loop;
+       end T;
+    end Simple_Sync_Pkg;
+
+As soon as the package is :ada:`with`'ed for the main procedure, the task
+``T`` defined in the package will be part of the main application. For
+example:
+
+.. code-block:: ada
+
+    --% run_file: Test_Simple_Sync_Pkg.adb
+    --% make_flags: -gnaty
+
+    with Simple_Sync_Pkg;
+
+    procedure Test_Simple_Sync_Pkg is
+    begin
+       null;
+       --  Will wait here until all tasks have terminated
+    end Test_Simple_Sync_Pkg;
+
+Again, as soon as the main application reaches its end, it will
+synchronize with task ``T`` from ``Simple_Sync_Pkg`` before finishing.
 
 Delay
------
+~~~~~
+
+A delay may be introduced by using the keyword :ada:`delay`. This will
+put the task to sleep for the amount of seconds specified in the delay
+statement. For example:
+
+.. code-block:: ada
+
+    --% run_file: Show_Delay.adb
+    --% make_flags: -gnaty
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Delay is
+
+       task T;
+
+       task body T is
+       begin
+          for I in 1 .. 10 loop
+             Put_Line ("hello");
+             delay 1.0;
+             --    ^ Wait 1.0 seconds
+          end loop;
+       end T;
+    begin
+       null;
+    end Show_Delay;
+
+In this example, we're making the task ``T`` wait one second after each
+time it displays the "hello" message.
 
 Synchronization: rendez-vous
-----------------------------
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In the previous examples, the only kind of synchronization we've seen was
+the one that happens automatically at the end of the main application. In
+addition to that, it is possible to define custom synchronization points
+(or entry points) using the keyword :ada:`entry`.
+
+In the task implementation, we shall define in which part of the task the
+entries will be accepted by using the keyword :ada:`accept`. A task will
+process its statements until it reaches an :ada:`accept` statement. At
+this point, it will wait for the main task or application to synchronize
+with the task. In other words:
+
+- the task will :ada:`accept` a synchronization to happen and it will be
+  waiting in that :ada:`entry` point.
+
+- the main application will call the :ada:`entry` point in order to
+  synchronize with the task.
+
+This synchronization between an application and a task via an :ada:`entry`
```

Comment:
It's actually a synchronization between a task and another task. It's fine to explain in the context of an application where one of those tasks is the main task, however you should be very clear about how general the concept is.